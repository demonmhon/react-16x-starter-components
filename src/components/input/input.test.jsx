import React from 'react';
import { shallow, mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Input from './index';

describe('Input component', () => {
  describe('Input', () => {
    test('it matches the snapshot', () => {
      const component = create(<Input />);
      expect(component.toJSON()).toMatchSnapshot();
    });

    test('has the provided class name', () => {
      const className = 'the-custom-css-class';
      const component = shallow(<Input className={className} />);
      expect(component.hasClass(className)).toBe(true);
    });

    describe('attributes', () => {
      test('value set correctly', () => {
        const props = {
          value: 'input value 1234',
        };
        const component = mount(<Input {...props} />);
        expect(component.find('input').props().value).toEqual(props.value);
      });

      test('disabled set correctly', () => {
        const props = {
          disabled: true,
        };
        const component = mount(<Input {...props} />);
        expect(component.find('input').props().disabled).toEqual(true);
      });
    });

    describe('onChange()', () => {
      test('called', () => {
        const props = {
          onChange: jest.fn(),
        };
        const component = mount(<Input {...props} />);
        component
          .find('input')
          .simulate('change', { target: { value: 'new-value' } });
        expect(props.onChange).toHaveBeenCalledTimes(1);
      });

      test('not called if disabled', () => {
        const props = {
          onChange: jest.fn(),
        };
        const component = mount(<Input {...props} />);
        component
          .find('input')
          .simulate('change', { target: { value: 'new-value' } });
        expect(props.onChange).toHaveBeenCalledTimes(1);
      });
    });
  });

  describe('Input.Textarea', () => {
    describe('onChange()', () => {
      test('called', () => {
        const props = {
          onChange: jest.fn(),
        };
        const component = mount(<Input.Textarea {...props} />);
        component
          .find('textarea')
          .simulate('change', { target: { value: 'new-value' } });
        expect(props.onChange).toHaveBeenCalledTimes(1);
      });

      test('not called if disabled', () => {
        const props = {
          onChange: jest.fn(),
        };
        const component = mount(<Input.Textarea {...props} />);
        component
          .find('textarea')
          .simulate('change', { target: { value: 'new-value' } });
        expect(props.onChange).toHaveBeenCalledTimes(1);
      });
    });
  })
});
