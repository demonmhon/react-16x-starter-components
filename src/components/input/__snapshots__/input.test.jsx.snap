// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Input it matches the snapshot 1`] = `
<div
  className="starter-input  starter-input--type-text"
>
  <label>
    <input
      onChange={[Function]}
      onKeyDown={[Function]}
      type="text"
      value=""
    />
  </label>
</div>
`;
